leftwm/src/config/keybind.rs

```diff
@@ -0,0 +1,169 @@
+                leftwm_core::Command::FocusWindowTop(if self.value.is_empty() {
+                    false
+                } else {
+                    bool::from_str(&self.value)
+                        .context("invalid boolean value for FocusWindowTop")?
+                })
+            }
+            BaseCommand::FocusWorkspaceNext => leftwm_core::Command::FocusWorkspaceNext,
+            BaseCommand::FocusWorkspacePrevious => leftwm_core::Command::FocusWorkspacePrevious,
+            BaseCommand::MoveToTag => leftwm_core::Command::SendWindowToTag {
+                window: None,
+                tag: usize::from_str(&self.value)
+                    .context("invalid index value for SendWindowToTag")?,
+            },
+            BaseCommand::MoveToLastWorkspace => leftwm_core::Command::MoveWindowToLastWorkspace,
+            BaseCommand::MoveWindowToNextWorkspace => {
+                leftwm_core::Command::MoveWindowToNextWorkspace
+            }
+            BaseCommand::MoveWindowToPreviousWorkspace => {
+                leftwm_core::Command::MoveWindowToPreviousWorkspace
+            }
+            BaseCommand::MouseMoveWindow => leftwm_core::Command::MouseMoveWindow,
+            BaseCommand::NextLayout => leftwm_core::Command::NextLayout,
+            BaseCommand::PreviousLayout => leftwm_core::Command::PreviousLayout,
+            BaseCommand::SetLayout => leftwm_core::Command::SetLayout(
+                Layout::from_str(&self.value)
+                    .context("could not parse layout for command SetLayout")?,
+            ),
+            BaseCommand::RotateTag => leftwm_core::Command::RotateTag,
+            BaseCommand::IncreaseMainWidth => leftwm_core::Command::IncreaseMainWidth(
+                i8::from_str(&self.value).context("invalid width value for IncreaseMainWidth")?,
+            ),
+            BaseCommand::DecreaseMainWidth => leftwm_core::Command::DecreaseMainWidth(
+                i8::from_str(&self.value).context("invalid width value for DecreaseMainWidth")?,
+            ),
+            BaseCommand::SetMarginMultiplier => leftwm_core::Command::SetMarginMultiplier(
+                f32::from_str(&self.value)
+                    .context("invalid margin multiplier for SetMarginMultiplier")?,
+            ),
+            BaseCommand::UnloadTheme => leftwm_core::Command::Other("UnloadTheme".into()),
+            BaseCommand::LoadTheme => {
+                leftwm_core::Command::Other(format!("LoadTheme {}", ensure_non_empty!(&self.value)))
+            }
+        };
+
+        Ok(leftwm_core::Keybind {
+            command,
+            modifier: self
+                .modifier
+                .as_ref()
+                .unwrap_or(&"Mod4".into())
+                .clone()
+                .into(),
+            key: self.key.clone(),
+        })
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Hash, Eq, PartialEq)]
+#[serde(untagged)]
+pub enum Modifier {
+    Single(String),
```

Comment:
I know this is good with TOML, but will this create issues for other formats, e.g. KDL or RON?

Reply:
This should be okay, AFAIK RON implements all of serdes functionality.